test/parallel/test-http-server-de-chunked-trailer.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+const common = require('../common');
+
+// This test ensures that a Trailer header is set only when a chunked transfer
+// encoding is used.
+
+const assert = require('assert');
+const http = require('http');
+
+const server = http.createServer(common.mustCall(function(req, res) {
+  res.setHeader('Trailer', 'baz');
+  const trailerInvalidErr = {
+    code: 'ERR_HTTP_TRAILER_INVALID',
+    message: 'Trailers are invalid with this transfer encoding',
+    type: Error
+  };
+  assert.throws(() => res.writeHead(200, {'Content-Length': '24'}),
+                common.expectsError(trailerInvalidErr));
+  res.removeHeader('Trailer');
+  res.end('ok');
+}));
+server.listen(0, common.mustCall(() => {
+  http.get({ port: server.address().port }, common.mustCall((res) => {
+    server.close();
```

Comment:
Minor nit: you might add a little extra code here to verify that the response is what you expect (both `statusCode` and body content). In doing so, you'd need to make sure to adjust the `Content-Length` value to 2.